App.js

```diff
@@ -1,86 +1,91 @@
 import Constants from 'expo-constants';
 import createWebNavigator from './createWebNavigator';
+import * as Linking from 'expo-linking';
 
 const chevronStyle = {width: 30, height: 30, resizeMode: 'contain', marginRight: 10};
 const chatTitleStyle = {fontSize: 18, fontWeight: 'bold'};
 const titleStyle = {fontSize: 24, fontWeight: 'bold', flex: 1};
 
 const config = {
-  initialRouteName: '',
+  initialRouteName: 'LeftHandNav',
   screens: {
-    Chat: 'r/:id?',
+    Chat: { 
+      path: 'r/:id?', 
+      parse: { id: (id) => parseInt(id) } 
+    },
     LeftHandNav: '',
     SettingsStack: {
+      path: 'settings',
       screens: {
-        Settings: 'settings',
-        About: 'settings/about',
+        Settings: '',
+        About: 'about',
       },
     },
     Search: 'search'
   },
 };
 
-const linking = {
-  prefixes: ['http://localhost', 'https://navigation-test-lime.vercel.app'],
-  config,
-  getStateFromPath(path, cfg) {
-    const state = getStateFromPath(path, cfg);
-    console.log('Get state from path: ', state);
+const stripKeysFromNavigationState = ({key, stale, routeNames, routes, state, ...rest}) => {
+  return {
+    ...rest, 
+    routes: routes && routes.map(stripKeysFromNavigationState),
+    state: state && stripKeysFromNavigationState(state)
+  }
+}
 
-    // We need to add the LHN if it does not exist
-    if (!_.find(state.routes, r => r.name === 'LeftHandNav')) {
-      state.routes.splice(0, 0, {name: 'LeftHandNav'});
+// TODO: make this function work better or find another way!
+const fixState = (state) => {
+    // we don't want to add Chat route for small screens
+    if (checkIsSmallScreen(Dimensions.get('window').width)) {
+      return 
     }
-
     if (!_.find(state.routes, r => r.name === 'Chat')) {
       state.routes.splice(1, 0, {name: 'Chat', params: {id: 1}});
     }
+    return state;
+}
 
+const linking = {
+  prefixes: ['http://localhost', 'https://navigation-test-lime.vercel.app', Linking.createURL('/')],
+  config,
+  getStateFromPath(path, cfg) {
+    const state = getStateFromPath(path, cfg);
+    fixState(state)
     return state;
   }
 };
 
-const isSmallScreenWidth = Dimensions.get('window').width <= 800;
-
-/**
- * By default the back handler will pop. We want it to go "back" instead of "up". All screens that are inside a nested navigator
- * except the root screen of that nested navigator will need to implement this.
- */
-function WithCustomBackBehavior(props) {
-  useFocusEffect(() => {
-    const subscription = BackHandler.addEventListener('hardwareBackPress', () => {
-      props.navigation.popToTop();
-    });
-
-    return () => subscription.remove();
-  });
+const checkIsSmallScreen = (width) => width <= 800;
 
-  return props.children;
+const useFocusedRouteParams = () => {
+  const state = useNavigationState(state => state)
+  return state.routes?.[state.index]?.params
 }
 
 function LeftHandNav({navigation}) {
+  const focusedRouteParams = useFocusedRouteParams();
```

Comment:
@adamgrzybowski I think in the App itself we would not want to make the buttons `disabled` that user cannot click on it, I would ay we should allow them to click on it but we would intercept the navigation and do nothing if the route is same as we are at.

Would we need to add a wrapper over the push similarly as we have the wrapped here https://github.com/Expensify/App/blob/a171133dbbda4552366511fc2acedec1816e6137/src/libs/Navigation/Navigation.js#L143-L176